backend/cmd/headlamp.go

```diff
@@ -1465,6 +1472,146 @@ func (c *HeadlampConfig) handleClusterRequests(router *mux.Router) {
 	handleClusterAPI(c, router)
 }
 
+// handleMe returns user info extracted from the per-cluster OIDC cookie so the UI can display the current user.
+// Claim discovery is controlled via the Headlamp configuration (HEADLAMP_CONFIG_ME_* env / --me-*-path flags),
+// which accept comma-separated JMESPath expressions to support different identity providers and nested claim layouts.
+func (c *HeadlampConfig) handleMe(w http.ResponseWriter, r *http.Request) {
+	clusterName := mux.Vars(r)["clusterName"]
+	if clusterName == "" {
+		http.Error(w, "cluster not specified", http.StatusBadRequest)
+		return
+	}
+
+	token, err := auth.GetTokenFromCookie(r, clusterName)
+	if err != nil || token == "" {
+		http.Error(w, "unauthorized", http.StatusUnauthorized)
+		return
+	}
+
+	parts := strings.SplitN(token, ".", 3)
+	if len(parts) != 3 || parts[1] == "" {
+		http.Error(w, "invalid token", http.StatusUnauthorized)
+		return
+	}
+
+	claims, err := auth.DecodeBase64JSON(parts[1])
+	if err != nil {
+		http.Error(w, "invalid token claims", http.StatusUnauthorized)
+		return
+	}
+
+	usernamePaths := c.meUsernamePaths
+	if strings.TrimSpace(usernamePaths) == "" {
+		usernamePaths = cfg.DefaultMeUsernamePath
+	}
+
+	emailPaths := c.meEmailPaths
+	if strings.TrimSpace(emailPaths) == "" {
+		emailPaths = cfg.DefaultMeEmailPath
+	}
+
+	groupsPaths := c.meGroupsPaths
+	if strings.TrimSpace(groupsPaths) == "" {
+		groupsPaths = cfg.DefaultMeGroupsPath
+	}
+
+	username := stringValueFromJMESPaths(claims, usernamePaths)
+	email := stringValueFromJMESPaths(claims, emailPaths)
+	groups := stringSliceFromJMESPaths(claims, groupsPaths)
+
+	// Prevent caching to avoid 304 Not Modified responses from intermediaries/dev server
+	w.Header().Set("Content-Type", "application/json")
+	w.Header().Set("Cache-Control", "no-store, no-cache, must-revalidate, private")
+	w.Header().Set("Pragma", "no-cache")
+	w.Header().Set("Expires", "0")
+	w.Header().Del("ETag")
+	_ = json.NewEncoder(w).Encode(map[string]interface{}{
+		"username": username,
+		"email":    email,
+		"groups":   groups,
+	})
+}
+
+// stringValueFromJMESPaths tries multiple comma-separated JMESPath expressions and returns the first string result.
+func stringValueFromJMESPaths(payload map[string]interface{}, pathCSV string) string {
```

Comment:
I don't see stringValueFromJMESPaths specific unit tests. Do the other tests in this PR cover it?